src/main/java/org/apache/cassandra/sidecar/cluster/instance/InstanceMetadataImpl.java

```diff
@@ -188,6 +203,29 @@ public Builder delegate(CassandraAdapterDelegate delegate)
             return update(b -> b.delegate = delegate);
         }
 
+        /**
+         * Builds instance specific {@link com.codahale.metrics.MetricRegistry} given the global registry name.
+         *
+         * @param registryName global {@link com.codahale.metrics.MetricRegistry} name
+         * @return a reference to this Builder
+         */
+        public Builder globalMetricRegistryName(String registryName)
+
+        {
+            Objects.requireNonNull(registryName);
+
+            String instanceRegistryName = instanceRegistryName(registryName);
+            MetricRegistry instanceMetricRegistry = SharedMetricRegistries.getOrCreate(instanceRegistryName);
+            InstanceMetricsImpl instanceMetrics = new InstanceMetricsImpl(instanceMetricRegistry);
+
+            return update(b -> b.metrics = instanceMetrics);
+        }
+
+        private String instanceRegistryName(String globalRegistryName)
+        {
+            return globalRegistryName + "_" + id;
```

Comment:
since `id` is an`int`. Same could happen if no id was set before build. Maybe we should convert it into an integer and check if set, before setting registry

Reply:
Setting registry before building makes sense. Modified this